browser-specific-failures.js

```diff
@@ -0,0 +1,386 @@
+
+  // Verify that we have data for the fetched runs in the wpt-results repo.
+  console.log('Getting local set of run ids from repo');
+  let before = Date.now();
+  let localRunIds = await lib.results.getLocalRunIds(repo);
+  let after = Date.now();
+  console.log(`Found ${localRunIds.size} ids (took ${after - before} ms)`);
+
+  let hadErrors = false;
+  for (const [date, runs] of alignedRuns.entries()) {
+    for (const run of runs) {
+      if (!localRunIds.has(run.id)) {
+        // If you see this, you probably need to run git-write.js or just update
+        // your wpt-results.git repo; see the README.md.
+        console.log(`ERROR: run ${run.id} missing from local git repo (date=${date})`);
+        hadErrors = true;
+      }
+    }
+  }
+  if (hadErrors) {
+    throw new Error('Missing data for some runs (see errors logged above). ' +
+        'Try running "git fetch --all --tags" in wpt-results/');
+  }
+
+  // Load the test result trees into memory; creates a list of recursive tree
+  // structures: tree = { trees: [...], tests: [...] }. Each 'tree' represents a
+  // directory, each 'test' is the results from a given test file.
+  console.log('Iterating over all runs, loading test results');
+  before = Date.now();
+  for (const runs of alignedRuns.values()) {
+    for (const run of runs) {
+      // Just in case someone ever adds a 'tree' field to the JSON.
+      if (run.tree)
+        throw new Error('Run JSON already contains "tree" field; code needs changed.');
+      run.tree = await lib.results.getGitTree(repo, run);
+    }
+  }
+  after = Date.now();
+  console.log(`Loading ${alignedRuns.size} sets of runs took ${after - before} ms`);
+
+  // We're ready to score the runs now!
+  console.log(`Calculating browser-specific failures for the runs (may take >5 minutes)`);
+  before = Date.now();
+  const dateToScores = new Map();
+  for (const [date, runs] of alignedRuns.entries()) {
+    // The SHA should be the same for all runs, so just grab the first.
+    const sha = runs[0].revision;
+    const scores = scoreBrowserSpecificFailures(runs, new Set(products));
+    dateToScores.set(date, {sha, scores});
+  }
+  after = Date.now();
+  console.log(`Done scoring (took ${after - before} ms)`);
+
+  // Finally, time to dump stuff.
+  let outputFilename = flags.get('output');
+  if (!outputFilename) {
+    outputFilename = experimental ?
+        'experimental-browser-specific-failures.csv' :
+        'stable-browser-specific-failures.csv';
+  }
+
+  console.log(`Writing data to ${outputFilename}`);
+  let data = `sha,date,${products.join(',')}\n`;
+  // ES6 maps iterate in insertion order, and we initially inserted in date
+  // order, so we can just iterate |dateToScores|.
+  for (const [date, shaAndScores] of dateToScores) {
+    let sha = shaAndScores.sha;
+    let scores = shaAndScores.scores;
+    if (!scores) {
+      console.log(`ERROR: ${date} had no scores`);
```

Comment:
Why not `console.error`? Or just `console.warn` since the problem is ignored.

Reply:
Looks like this was closed without any code change?